torch/distributed/fsdp/symbolic_trace.py

```diff
@@ -0,0 +1,202 @@
+import functools
+import torch
+from dataclasses import dataclass
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Set,
+    Tuple,
+)
+
+
+__all__ = ["TracingConfig"]
+
+
+@dataclass
+class TracingConfig:
+    """
+    Configurations used in ParamExecOrderWrapPolicy for symbolic tracing of a model.
+    tracer: An instance of torch.fx.Tracer that will be used to perform symbolic
+        tracing. tracer is default to be torch.fx.Tracer(), but can also be instance
+        of some child class of torch.fx.Tracer. For example, for hugginface transformer
+        based models, one may want to use HFTracer:
+        https://github.com/huggingface/transformers/blob/6dd00f6bd49141ef4f26ed1d1c555bc0fe109ea8/src/transformers/utils/fx.py#L636
```

Comment:
Let's make links reference in doc instead of plain text URLs

Reply:
Currently, HFTracer doesn't have a doc link in the transformers package, not sure if there are better ways to show the link

Reply:
see this: https://github.com/pytorch/pytorch/blob/ea987086fce93d544db72706928c425e339b8bce/torch/distributed/fsdp/fully_sharded_data_parallel.py#L443-L444